apps/src/templates/teacherDashboard/ScriptSelector.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component, PropTypes } from 'react';
+import _ from 'lodash';
+
+// TODO: Can/should we share any logic with AssignmentSelector?
```

Comment:
The answer might be that they're just different.

This is a little confusing in that they use slightly difference sources of data for their list of scripts. AssignmentSelector uses a set of assignments that have been slightly processed, to be able to differentiate scripts and courses. In part this involves giving each one an "assignment id". That said, both have the same set of fields used for grouping I believe.